regression caused by WEXITSTATUS: `nim cpp compiler/nim.nim` fails on OSX

/cc @alaviss 
this regression was probably caused by https://github.com/nim-lang/Nim/pull/10222
```
nim cpp compiler/nim.nim

Error: execution of an external compiler program 'clang++ -c  -w  -I/Users/timothee/git_clone/nim/Nim_devel/lib -I/Users/timothee/git_clone/nim/Nim_devel/compiler -o /tmp/d11/stdlib_os.cpp.o /tmp/d11/stdlib_os.cpp' failed with exit code: 1

/tmp/d11/stdlib_os.cpp:3079:8: error: cannot take the address of an rvalue of type 'int'
        T1_ = WEXITSTATUS(system(nimToCStringConv(command)));
              ^           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/sys/wait.h:144:26: note: expanded from macro 'WEXITSTATUS'
#define WEXITSTATUS(x)  ((_W_INT(x) >> 8) & 0x000000ff)
                          ^      ~
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/sys/wait.h:131:28: note: expanded from macro '_W_INT'
#define _W_INT(w)       (*(int *)&(w))  /* convert union wait to int */
                                 ^ ~
```

## note
to avoid further such regression, we should at least compile some compiler sources (eg maybe just `nim cpp compiler/nim.nim`)  on CI / OSX

* note: not sure if it's just OSX or other platforms


## relevant snippet:
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/usr/include/sys/wait.h:
```
/*
 * Macros to test the exit status returned by wait
 * and extract the relevant values.
 */
#if defined(_POSIX_C_SOURCE) && !defined(_DARWIN_C_SOURCE)
#define	_W_INT(i)	(i)
#else
#define	_W_INT(w)	(*(int *)&(w))	/* convert union wait to int */
#define	WCOREFLAG	0200
#endif /* (_POSIX_C_SOURCE && !_DARWIN_C_SOURCE) */

/* These macros are permited, as they are in the implementation namespace */
#define	_WSTATUS(x)	(_W_INT(x) & 0177)
#define	_WSTOPPED	0177		/* _WSTATUS if process is stopped */

/*
 * [XSI] The <sys/wait.h> header shall define the following macros for
 * analysis of process status values
 */
#if __DARWIN_UNIX03
#define WEXITSTATUS(x)	((_W_INT(x) >> 8) & 0x000000ff)
#else /* !__DARWIN_UNIX03 */
#define WEXITSTATUS(x)	(_W_INT(x) >> 8)
#endif /* !__DARWIN_UNIX03 */
```

--- a/lib/pure/os.nim
+++ b/lib/pure/os.nim
@@ -1297,7 +1297,9 @@ proc execShellCmd*(command: string): int {.rtl, extern: "nos$1",
   ## the process has finished. To execute a program without having a
   ## shell involved, use the `execProcess` proc of the `osproc`
   ## module.
-  when defined(posix):
+  when defined(macosx):
+    result = c_system(command) shr 8
+  elif defined(posix):
     result = WEXITSTATUS(c_system(command))
   else:
     result = c_system(command)

--- a/tests/stdlib/t10231.nim
+++ b/tests/stdlib/t10231.nim
@@ -1,0 +1,13 @@
+discard """
+  target: cpp
+  action: run
+  exitcode: 0
+"""
+
+import os
+
+if paramCount() == 0:
+  # main process
+  doAssert execShellCmd(getAppFilename().quoteShell & " test") == 1
+else:
+  quit 1

